From d0009daed22401b3cd9ed4c0888499060c972b23 Mon Sep 17 00:00:00 2001
From: Zhang Jiejing <jiejing.zhang@freescale.com>
Date: Wed, 19 Jan 2011 16:29:35 +0800
Subject: [PATCH 09/38] ENGR00138188 runtime: add a fake 3G modem path

Add a fake port in runtime check rild.since if return NULL to ril-reference,
it will cause rild init failed, so the framwork can't connect to rild,
and will hold a wake lock, this will cause system can't into suspend,
it will in early suspend 15 seconds.

so add a fake port to fix this.

Signed-off-by: Zhang Jiejing <jiejing.zhang@freescale.com>
---
 reference-ril/runtime_port.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/reference-ril/runtime_port.c b/reference-ril/runtime_port.c
index f262166..2ce32ab 100644
--- a/reference-ril/runtime_port.c
+++ b/reference-ril/runtime_port.c
@@ -33,6 +33,12 @@
 
 int current_modem_type = UNKNOWN_MODEM;
 
+#define FAKE_PORT "/dev/ttyFAKEPort"
+/* Rild need a fake port to pass continue init job,
+ * return a fake port make it runable.
+ * Or the system will enter 15s in early suspend.
+ */
+
 struct modem_3g_device {
 	const char *idVendor;
 	const char *idProduct;
@@ -195,7 +201,7 @@ const char *runtime_3g_port_device(void)
 
 	device = find_matched_device();
 	if (device == NULL)
-		return NULL;
+		return FAKE_PORT;
 
 	/* Set gobal modem type. */
 	current_modem_type = device->type;
@@ -211,7 +217,7 @@ const char *runtime_3g_port_data(void)
 
 	device = find_matched_device();
 	if (device == NULL)
-		return NULL;
+		return FAKE_PORT;
 	return device->dataport;
 }
 
-- 
1.8.0

